ovn_k8s/processor/policy_processor.py

```diff
@@ -28,34 +30,192 @@ def __init__(self, pool):
+        pods = kubernetes.get_pods_by_namespace(
+            variables.K8S_API_SERVER,
+            namespace=namespace,
+            pod_selector=pod_selector)
+        for pod in pods:
+            pod_id = pod['metadata']['uid']
+            affected_pods[pod_id] = pod
+            pod_events.setdefault(pod_id, []).append(event)
+
+        if event.event_type == 'DELETED':
+            # Remove pseudo ACLs for policy and address sets for all of
+            # its rules
+            self.mode.destroy_address_set(namespace, policy_data)
+            self.mode.remove_pseudo_acls(policy_id)
+            log.dbg("Policy: %s (Namespace: %s): pseudo ACLs and address sets"
+                    "destroyed" % (policy_id, namespace))
+            return
+        else:
+            # As there is no MODIFIED event for policies, if we end up here
+            # we are hanndling an ADDED event
+            self.mode.create_address_set(namespace, policy_data)
+            self.mode.build_pseudo_acls(policy_data)
+            self.mode.add_pods_to_policy_address_sets(policy_data)
+            log.dbg("Policy: %s (Namespace: %s): pseudo ACLs and address sets"
+                    "created" % (policy_id, namespace))
+
+    def _apply_pod_ns_acls(self, pod_data, pod_events):
+        ns_events = [event for event in pod_events if
+                     isinstance(event, processor.NSEvent)]
+        if not ns_events:
+            return
+        namespace = pod_data['metadata']['namespace']
+        pod_name = pod_data['metadata']['name']
+        pod_id = pod_data['metadata']['uid']
+        log.dbg("Pod: %s (Namespace: %s): Applying ACL changes for "
+                "namespace events" % (pod_name, namespace))
+        # In some rare cases, there could multiple namespace events. The
+        # ns_events list reports then in the same order in which they were
+        # detected and is therefore reliable. We only check first and last as
+        # multiple transitions might nullify each other
+        if ns_events[0] == ns_events[-1]:
+            # Single event
+            ns_data = ns_events[-1].metadata
+            isolated_final = ns_data.get('custom', {}).get('isolated', False)
+        else:
+            ns_data_final = ns_events[-1].metadata
+            isolated_final = ns_data_final.get(
+                'custom', {}).get('isolated', False)
+            ns_data_initial = ns_events[0].metadata
+            isolated_initial = ns_data_initial.get(
+                'custom', {}).get('isolated', False)
+            if isolated_final == isolated_initial:
+                # Nothing actually changed
+                log.dbg("Pod: %s (Namespace: %s): initial and final "
+                        "namespace isolation status are identical (%s) - "
+                        "no processing necessary" %
+                        (pod_name. namespace, isolated_final))
+                return
+        self.mode.remove_acls(pod_id)
```

Comment:
adding a TODO comment for now, will push improvements like this in different PRs.